libethereum/Executive.cpp

```diff
@@ -459,7 +480,63 @@ void Executive::finalize()
 	{
 		m_res->gasUsed = gasUsed();
 		m_res->excepted = m_excepted; // TODO: m_except is used only in ExtVM::call
-		m_res->newAddress = m_newAddress;
+		m_res->newAddress = m_orig.address;
 		m_res->gasRefunded = m_ext ? m_ext->sub.refunds : 0;
 	}
 }
+
+void Executive::revert()
+{
+	if (m_ext)
+		m_ext->sub.clear();
+
+	eth::revert(m_s, m_orig);
+	m_orig.address = {};
+}
+
+void dev::eth::revert(State& _state, AccountSnapshot const& _changes)
```

Comment:
It was used by both Executive and ExtVM at some point, but I guess it not any more. To be fixed soon...